src/crypto.rs

```diff
@@ -668,7 +668,7 @@ impl Deref for PrivKeyValue {
 }
 
 #[derive(PartialEq)]
-pub struct PubKeyValue(Vec<u8>);
+pub struct PubKeyValue(pub Vec<u8>);
```

Comment:
This shouldn't be public because I tried to encode into the types the logic that "you should never be changing a public key's value."